web/src/message_reminder.ts

```diff
@@ -0,0 +1,38 @@
+import * as blueslip from "./blueslip.ts";
+import * as channel from "./channel.ts";
+import * as feedback_widget from "./feedback_widget.ts";
+import {$t} from "./i18n.ts";
+import * as timerender from "./timerender.ts";
+
+export function set_message_reminder(send_at_time: number, message_id: number): void {
+    channel.post({
+        url: "/json/reminders",
+        data: {
+            message_id,
+            scheduled_delivery_timestamp: send_at_time,
+        },
+        success(): void {
+            const populate: (element: JQuery) => void = ($container) => {
+                $container.html(
+                    $t(
+                        {defaultMessage: "Your reminder has been scheduled for {translated_time}."},
+                        {
+                            translated_time: timerender.get_full_datetime(
+                                new Date(send_at_time * 1000),
+                                "time",
+                            ),
+                        },
+                    ),
+                );
+            };
+            const title_text = $t({defaultMessage: "Reminder scheduled"});
+            feedback_widget.show({
+                populate,
+                title_text,
+            });
+        },
+        error(xhr: JQuery.jqXHR): void {
+            blueslip.error(channel.xhr_error_message("Error setting reminder", xhr));
```

Comment:
s/setting/scheduling seems correct here. Also, do we need that string at all, or can we just use the server's error message? I kinda don't want to have an extra string here, which we'd have to translate, given it's not likely to fail.